server/s3_object_metadata.cc

```diff
@@ -1695,18 +1697,22 @@ bool S3ObjectMetadata::check_bucket_replication_policy(
                                               new_rule_priority);
 
     } else {
-      s3_log(S3_LOG_DEBUG, request_id, "Filter does not matched.\n");
-      return false;
+      continue;
     }
   }
-  // Added this log for demo purpose.Can be remove later.
-  for (auto dest : destination_bucket_map) {
-    std::map<std::string, std::string> dest_map = dest.second;
-
-    s3_log(S3_LOG_INFO, stripped_request_id, " %s ::[%s, %s ]\n",
-           dest.first.c_str(),
-           dest_map["DeleteMarkerReplicationStatus"].c_str(),
-           dest_map["Priority"].c_str());
+  if (destination_bucket_map.empty()) {
+    s3_log(S3_LOG_DEBUG, request_id, "Filter does not matched.\n");
+    return false;
+  } else {
+    // Added this log for demo purpose.Can be remove later.
+    for (auto dest : destination_bucket_map) {
+      std::map<std::string, std::string> dest_map = dest.second;
+
+      s3_log(S3_LOG_INFO, stripped_request_id, " %s ::[%s, %s ]\n",
+             dest.first.c_str(),
+             dest_map["DeleteMarkerReplicationStatus"].c_str(),
+             dest_map["Priority"].c_str());
+    }
   }
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Exit\n", __func__);
   return true;
```

Comment:
We can keep the original code, and then
```
  return !destination_bucket_map.empty();

Reply:
Because the "for(){}" block will be be executed if the `destination_bucket_map` is empty.

Reply:
Done.